### PR TITLE
Add verification test for event listener data

### PR DESCRIPTION
**Pull request by nategraf**
Adds a test which can verify that objects are properly received by an EventListener. This provides verification beyond the current smoke tests, but stops short of verifying what is written to a file

References: #11567

### CODE REVIEW

**Comment by nategraf** [comment]
@dotnet-bot test Ubuntu pri1 please

File: tests/src/tracing/eventsourcewithlistener/EventSourceWithListener.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
+using System;
```
Comment (jkotas) [inline tests/src/tracing/eventsourcewithlistener/EventSourceWithListener.cs]: License

Comment (nategraf) [reply tests/src/tracing/eventsourcewithlistener/EventSourceWithListener.cs]: Thanks for noticing that

Comment (nategraf) [reply tests/src/tracing/eventsourcewithlistener/EventSourceWithListener.cs]: Thanks for noticing that

**Comment by brianrob** [comment]
@nategraf, I think this change looks fine, but I do want to understand the overall goal.  We do have a number of tests like this in CoreFx that test the EventSource --> EventListener pipeline.  That's a nice place to put the managed surface area test cases because they can then run against any runtime.

If the goal of this is to get us closer to being able to test the input/output of the EventPipe, then this feels like the right place to put this test.  Can you confirm that you are basically preparing to make this validate the end-to-end for the EventPipe?

**Comment by nategraf** [comment]
Yes, so this test is basically a halfway point to testing the full EventPipe system as the data comes out of the file. I did not realise this functionality is already tested in CoreFX. I am going to close this PR and open a new one when I have a test that acts on the final file output of EventPipe

**Pull request closed without merging**